File: descend_calc/descend_calc.js
const readline = require('readline');

async function get_input(msg)
{
    const rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    });

    while (true)
    {
      const line = await new Promise(resolve =>
        rl.question(msg, resolve)
      );

      const num = parseInt(line.trim(), 10);
      if (!isNaN(num))
      {
        rl.close();
        return num;
      }
  }
}

async function main()
{
    const current_altitude = await get_input("Current altitude: ")
    const airport_altitude = await get_input("Airport altitude: ")

    const altitude_to_lose = ((current_altitude - airport_altitude) / 100.0) / 3.0

    const current_speed = await get_input("Current speed: ")
    const avg_dsc_speed = await get_input("Average descend speed: ")

    const speed_knots = (current_speed - avg_dsc_speed) / 10.0

    const wind_speed = await get_input("Wind speed: ") / 10.0
    const tail_wind = await get_input("Tail or head wind [1 for tail | 0 for head]: ")
    
    const descend_distance = altitude_to_lose + speed_knots + (tail_wind == 1 ? wind_speed : -wind_speed)

    console.log("------------------------------");
    console.log("Altitude to lose nm: " + altitude_to_lose);
    console.log("Speed nm: " + speed_knots);
    console.log("Wind  nm: " + wind_speed);
    console.log("Total distance: " + descend_distance);
}

main()

File: descend_calc/descend_calc.swift
func get_input(msg: String) -> Int
{
    while true
    {
        print(msg, terminator: "")
        if let num = Int(readLine()!)
        {
            return num
        }
    }
}

func main() -> Void
{
    let current_altitude = get_input(msg: "Current altitude: ")
    let airport_altitude = get_input(msg: "Airport altitude: ")

    let altitude_to_lose = (Float(current_altitude - airport_altitude) / 100.0 as Float) / 3.0 as Float

    let current_speed = get_input(msg: "Current speed: ")
    let avg_dsc_speed = get_input(msg: "Average descend speed: ")

    let speed_knots = Float(current_speed - avg_dsc_speed) / 10.0 as Float

    let wind_speed = Float(get_input(msg: "Wind speed: ")) / 10.0 as Float
    let tail_wind = get_input(msg: "Tail or head wind [1 for tail | 0 for head]: ")
    
    let descend_distance = altitude_to_lose + speed_knots + (tail_wind == 1 ? wind_speed : -wind_speed)

    print("------------------------------");
    print("Altitude to lose nm: \(altitude_to_lose)");
    print("Speed nm: \(speed_knots)");
    print("Wind  nm: \(wind_speed)");
    print("Total distance: \(descend_distance)");
}

main()